pkg/core/coretesting/resources.go

```diff
@@ -0,0 +1,49 @@
+package coretesting
+
+import (
+	"testing"
+
+	"github.com/klothoplatform/klotho/pkg/core"
+	"github.com/klothoplatform/klotho/pkg/graph"
+	"github.com/stretchr/testify/assert"
+)
+
+type (
+	StringDep = graph.Edge[string]
+
+	ResourcesExpectation struct {
+		Nodes []string
+		Deps  []StringDep
+
+		// OnlyContains only check whether the dag contains the `.Nodes` and `.Deps`. If false,
+		// checks full equality.
+		OnlyContains bool
```

Comment:
"Only" is an ambiguous word, and confused me in this usage. To me, this reads as "check that the dag Only Contains these elements", which is pretty much the opposite of what I think you meant ("Only check that the dag Contains these elements").

Maybe `PartialMatchAccepted` or such?